src/blocks/service_status.rs

```diff
@@ -141,27 +142,49 @@ impl SystemdDriver {
         let path = format!("/org/freedesktop/systemd1/unit/{encoded_service}");
 
         let proxy = UnitProxy::builder(&dbus_conn)
-            .path(path)
+            .path(path.clone())
             .error("Could not set path")?
             .build()
             .await
             .error("Failed to create UnitProxy")?;
 
+        let service_proxy = ServiceProxy::builder(&dbus_conn)
+            .path(path)
+            .error("Could not set path")?
+            .build()
+            .await
+            .error("Failed to create ServiceProxy")?;
+
         Ok(Self {
             active_state_changed: proxy.receive_active_state_changed().await,
             proxy,
+            service_proxy,
         })
     }
 }
 
 #[async_trait]
 impl Driver for SystemdDriver {
     async fn is_active(&self) -> Result<bool> {
-        self.proxy
+        let active_state = self
+            .proxy
             .active_state()
             .await
-            .error("Could not get active_state")
-            .map(|state| state == "active")
+            .error("Could not get active_state")?;
+
+        Ok(match &*active_state {
+            "active" => true,
+            "activating" => {
+                let service_type = self
+                    .service_proxy
+                    .type_()
+                    .await
+                    .error("Could not get service type")?;
```

Comment:
I read this property every time in case the underlying service changes.

Reply:
Under what circumstance would the `service_type` switch from `"oneshot"` to something else?

Reply:
The user changes the service file and calls `systemctl daemon-reload`.

Reply:
This is obviously a pretty weird edge case, but I'm imagining the following scenario:

1. User is developing a unit file.
2. They've wired up i3status-rs to monitor it.
3. They change it to a oneshot unit.
4. Suddenly it stops showing up in i3status-rs whenever they restart it.

(also, it's easy enough to just check every time)